django/db/migrations/operations/models.py

```diff
@@ -1274,3 +1274,42 @@ def describe(self):
     @property
     def migration_name_fragment(self):
         return "remove_%s_%s" % (self.model_name_lower, self.name.lower())
+
+
+class AlterConstraint(IndexOperation):
+    category = OperationCategory.ALTERATION
+    option_name = "constraints"
```

Comment:
Jacob can you expand why? This would be the only operation that we'd set this as default and I don't know why we'd want these operations to disappear on squash (but I'm not very familiar with `elidable`)

Reply:
Thanks Sarah, you're right I don't think we want these to disappear on squash because they would just trigger the autodetector again, but I was just wondering if there was a way to get them to optimize away with `optimizemigrations` if you have several in a row.

This seems to work just to demo the idea, instead of fiddling with elidable (not thoroughly tested though):
```diff
diff --git a/django/db/migrations/operations/models.py b/django/db/migrations/operations/models.py
index 1b7db95b5c..8bdc805b4f 100644
--- a/django/db/migrations/operations/models.py
+++ b/django/db/migrations/operations/models.py
@@ -1313,3 +1313,8 @@ class AlterConstraint(IndexOperation):
     @property
     def migration_name_fragment(self):
         return f"alter_{self.model_name_lower}_{self.constraint.name.lower()}"
+
+    def reduce(self, operation, app_label):
+        if isinstance(operation, AlterConstraint):
+            return [operation]
+        return False
```

Reply:
I think this works, can you check the test I've made?

Reply:
Thanks, looks great!